vm/src/dictdatatype.rs

```diff
@@ -412,40 +413,43 @@ enum LookupResult {
 /// - PyObjectRef -> arbitrary python type used as key
 /// - str -> string reference used as key, this is often used internally
 pub trait DictKey: IntoPyObject {
-    fn do_hash(&self, vm: &VirtualMachine) -> PyResult<HashValue>;
-    fn do_is(&self, other: &PyObjectRef) -> bool;
-    fn do_eq(&self, vm: &VirtualMachine, other_key: &PyObjectRef) -> PyResult<bool>;
+    fn key_hash(&self, vm: &VirtualMachine) -> PyResult<HashValue> {
+        let raw_hash = self.key_object_hash(vm)?;
+        let mut hasher = DefaultHasher::new();
+        raw_hash.hash(&mut hasher);
+        Ok(hasher.finish() as HashValue)
```

Comment:
is this intentionally hashed again due to some reason like security?

Reply:
Not sure, it does seem redundant. Does it work if you remove it?

Reply:
Oh, is this why you have the 2 different `key_hash` methods here/

Reply:
Yes, it work if I remove it. I kept it here because I wasn't sure it is intended or not.

And Yes, this is why we have duplicated hash. One for pure hash and another for wrapped key hash.

Reply:
would there be hashing quality problem? PyHash is basically 61bit integer.

Reply:
I don't think so, you only get as much entropy/variation out as you put in, so if you put an integer in that you know will only have 2^61 possible values then the output will still only be one of 2^61 possible outputs.

Reply:
Reasonable. Do you think removing this part is better?

Reply:
Yeah, I think it'd be best to do that; we already have mod_int for "normalizing" stuff to the same hash value.